packages/sdk-core/src/sourcemaps/DebugIdProvider.ts

```diff
@@ -0,0 +1,45 @@
+import { BacktraceStackTraceConverter } from '../modules/converter';
+import { DebugIdMapProvider } from './interfaces/DebugIdMapProvider';
+
+export const SOURCE_DEBUG_ID_VARIABLE = '_btDebugIds';
+
+export class DebugIdProvider {
+    private _fileDebugIds?: Record<string, string>;
+
+    constructor(
+        private readonly _stackTraceConverter: BacktraceStackTraceConverter,
+        private readonly _debugIdMapProvider?: DebugIdMapProvider,
```

Comment:
wouldn't it be better if sdk will pass debug id map instead of provider? The provider should always be defined, otherwise, there is no reason of generating debug id map. I see how you mock your test cases, but I believe in the real case scenario outside tests we won't pass debugIdMap to loadDebguids (what is the reason to pass it if we already have them?)

Reply:
I want to make this as transparent as possible. If for some reason we do not support debug IDs, it should not fail because of missing debug ID map provider.